aten/src/ATen/Context.cpp

```diff
@@ -74,4 +74,15 @@ cudaStream_t Context::getCurrentCUDAStream() const {
 }
 #endif
 
+int64_t Context::current_device() const {
+#if AT_CUDA_ENABLED()
+  int device;
+  cudaError_t err = cudaGetDevice(&device);
+  if (err == cudaSuccess) {
```

Comment:
Returning -1 in case of an error is a bit weird, because it's actually a meaningful value for device idx in PyTorch. Why not throw an error?

Reply:
I want the behavior for no GPUs to match the behavior for compiling without CUDA support. So that means both cases should return the same value or both cases should throw. In general, having these functions be safe to call in a context without CUDA is more useful than throwing exceptions.